sdk/cognitiveservices/azopenai/custom_client.go

```diff
@@ -239,7 +239,21 @@ func (client *Client) newError(resp *http.Response) error {
 }
 
 type clientData struct {
-	endpoint     string
-	baseEndpoint string
-	azure        bool
+	endpoint string
+	azure    bool
+}
+
+func getDeploymentID[T ChatCompletionsOptions | CompletionsOptions | EmbeddingsOptions | ImageGenerationOptions](v T) string {
+	switch a := any(v).(type) {
+	case ChatCompletionsOptions:
+		return a.DeploymentID
+	case CompletionsOptions:
+		return a.DeploymentID
+	case EmbeddingsOptions:
+		return a.DeploymentID
+	case ImageGenerationOptions:
+		return ""
+	default:
```

Comment:
You shouldn't need this.

Reply:
I wish I had a way to enforce that - the problem is that I can't type switch on the generic `T`, so I have to cast to `any` and do the type switch on that, which means that there's no checking that I haven't fallen through a default path.

I could change this to also return an 'err', which wouldn't be horrible, but this was just something I was trying to do to be expedient. :|